### PR TITLE
[20415] Update APIs in video performance tests 

### PR DESCRIPTION
**Pull request by elianalf**
## Description

This PR substitutes the old APIs with new ones in supports files for video performance tests.

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- N/A Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- N/A Any new/modified methods have been properly documented using Doxygen. 
- [x] Changes are ABI compatible. 
- [x] Changes are API compatible. 
- N/A New feature has been added to the `versions.md` file (if applicable).
- N/A New feature has been documented/Current behavior is correctly described in the documentation. 
    
- N/A Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: test/performance/video/VideoTestPublisher.cpp [hunk 1]
```diff
@@ -127,146 +169,169 @@ bool VideoTestPublisher::init(
...
+        {
+            mp_participant = DomainParticipantFactory::get_instance()->create_participant(
+                m_forcedDomain, participant_qos);
+        }
+        else
+        {
+            mp_participant = DomainParticipantFactory::get_instance()->create_participant(
+                pid % 230, participant_qos);
+        }
     }
```
Comment (JesusPoderoso) [inline test/performance/video/VideoTestPublisher.cpp:205]: I would keep the pid logic here (internal jenkins CI purposes)

File: test/performance/video/VideoTestSubscriber.hpp [hunk 2]
```diff
@@ -30,130 +31,211 @@
...
+
+    std::string m_sXMLConfigFile;
+    bool m_bRunning;
+
+    GstElement* pipeline;
+    GstElement* appsrc;
+    GstElement* sink;
+    GstElement* videoconvert;
+    guint source_id_;          // To control the GSource
+    GMainLoop* gmain_loop_;     // GLib's Main Loop
```
Comment (JesusPoderoso) [inline test/performance/video/VideoTestSubscriber.hpp:183]:
```suggestion
    guint source_id_;          // To control the GSource
    GMainLoop* gmain_loop_;    // GLib's Main Loop
```

**Review by JesusPoderoso — CHANGES_REQUESTED** [review]
A couple of NITS.
Please, consider addressing the linter issue.

File: test/performance/video/VideoTestPublisher.cpp [hunk 3]
```diff
@@ -127,146 +169,169 @@ bool VideoTestPublisher::init(
...
-    {
-        PParam.domainId = m_forcedDomain;
-    }
-    else
-    {
-        PParam.domainId = pid % 230;
-    }
-    PParam.rtps.properties = part_property_policy;
-    PParam.rtps.setName("video_test_publisher");
+    DomainParticipantQos participant_qos;
```
Comment (JesusPoderoso) [inline test/performance/video/VideoTestPublisher.cpp:174]: > I would keep the pid logic here (internal jenkins CI purposes)

```suggestion
    auto forced_domain = m_forcedDomain;
    if (m_forcedDomain >= 0)
    {
        forced_domain = m_forcedDomain;
    }
    else
    {
        forced_domain = pid % 230;
    }
    DomainParticipantQos participant_qos;
```

File: test/performance/video/VideoTestPublisher.cpp [hunk 4]
```diff
@@ -127,146 +169,169 @@ bool VideoTestPublisher::init(
...
-            }
+            mp_participant = DomainParticipantFactory::get_instance()->create_participant_with_profile(m_forcedDomain,
+                            participant_profile_name);
         }
         else
         {
-            mp_participant = Domain::createParticipant(participant_profile_name);
+            mp_participant = DomainParticipantFactory::get_instance()->create_participant_with_profile(
+                participant_profile_name);
         }
```
Comment (JesusPoderoso) [inline test/performance/video/VideoTestPublisher.cpp:191]: > I would keep the pid logic here (internal jenkins CI purposes)

remove displayed code, and add:

```
mp_participant = DomainParticipantFactory::get_instance()->create_participant_with_profile(forced_domain,
                            participant_profile_name);
```

File: test/performance/video/VideoTestPublisher.cpp [hunk 5]
```diff
@@ -127,146 +169,169 @@ bool VideoTestPublisher::init(
...
+        if (m_forcedDomain >= 0)
+        {
+            mp_participant = DomainParticipantFactory::get_instance()->create_participant(
+                m_forcedDomain, participant_qos);
+        }
+        else
+        {
+            mp_participant = DomainParticipantFactory::get_instance()->create_participant(
+                pid % 230, participant_qos);
+        }
```
Comment (JesusPoderoso) [inline test/performance/video/VideoTestPublisher.cpp:204]: > I would keep the pid logic here (internal jenkins CI purposes)

```suggestion
        mp_participant = DomainParticipantFactory::get_instance()->create_participant(
            forced_domain, participant_qos);
```

**Comment by JesusPoderoso** [comment]
This should be also addressed in the `test/performance/video/VideoTestSubscriber.cpp` file

**Comment by elianalf** [comment]
@richiprosima please test this

**Review by JesusPoderoso — APPROVED** [review]
LGTM

**Pull request merged**